packages/plugin-pnp/sources/PnpLinker.ts

```diff
@@ -297,6 +301,9 @@ export class PnpInstaller implements Installer {
       await xfs.chmodPromise(pnpDataPath, 0o644);
     }
 
+    if (pnpPath.main.endsWith(`.cjs`))
+      await xfs.writeFilePromise(pnpPath.esmLoader, getESMLoaderTemplate());
```

Comment:
Why this check? Note that one of the v3 item is to make the `.pnp.js` file be `.cjs` in every case. Will that work with this? 🤔

Reply:
It was to check if esm was enabled, i'll change it to use the correct check